vmcnet/updates/parse_config.py

```diff
@@ -519,3 +561,113 @@ def optimizer_apply(grad, params, optimizer_state, data, aux):
     )
 
     return update_param_fn, optimizer_state
+
+
+# >>>>>>>>>> from gg-min-sr-mom (modified) >>>>>>>>>>
+def get_proxsr_update_fn_and_state(
+    log_psi_apply: ModelApply[P],
+    params: P,
+    get_position_fn: GetPositionFromData[D],
+    update_data_fn: UpdateDataFn[D, P],
+    energy_data_val_and_grad: physics.core.ValueGradEnergyFn[P],
+    learning_rate_schedule: LearningRateSchedule,
+    optimizer_config: ConfigDict,
+    record_param_l1_norm: bool = False,
+    apply_pmap: bool = True,
+) -> Tuple[UpdateParamFn[P, D, optax.OptState], optax.OptState]:
+    """Get an update param function and initial state for stochastic reconfiguration.
+
+    Args:
+        log_psi_apply (Callable): computes log|psi(x)|, where the signature of this
+            function is (params, x) -> log|psi(x)|
+        params (pytree): params with which to initialize optimizer state
+        get_position_fn (Callable): function which gets the position array from the data
+        update_data_fn (Callable): function which updates data for new params
+        energy_data_val_and_grad (Callable): function which computes the clipped energy
+            value and gradient. Has the signature
+                (params, x)
+                -> ((expected_energy, auxiliary_energy_data), grad_energy),
+            where auxiliary_energy_data is the tuple
+            (expected_variance, local_energies, unclipped_energy, unclipped_variance)
+        learning_rate_schedule (Callable): function which returns a learning rate from
+            epoch number. Has signature epoch -> learning_rate
+        optimizer_config (ConfigDict): configuration for stochastic reconfiguration
+        descent_config (ConfigDict): configuration for the gradient descent-like method
+            used to apply the preconditioned updates
+        record_param_l1_norm (bool, optional): whether to record the L1 norm of the
+            parameters in the metrics. Defaults to False.
+        apply_pmap (bool, optional): whether to pmap the optimizer steps. Defaults to
+            True.
+
+    Raises:
+        ValueError: A non-supported descent type is requested. Currently only Adam and
+            SGD are supported.
+
+    Returns:
+        (UpdateParamFn, optax.OptState):
+        update param function with signature
+            (params, data, optimizer_state, key)
+            -> (new params, new state, metrics, new key), and
+        initial optimizer state
+    """
+    precondition_grad_fn = get_fisher_inverse_fn_proxsr(
```

Comment:
Can we rename this function as something like `get_proxsr_update_fn`? ProxSR doesn't really carry the meaning of the fisher inverse, although the fisher inverse is one step of the calculation